apps/examples/lcd_test/example_lcd.c

```diff
@@ -0,0 +1,264 @@
+	yres = vinfo.yres;
+	printf("xres : %d, yres:%d\n", xres, yres);
+	close(fd);
+
+	#if defined(CONFIG_LCD_PORTRAIT) || defined(CONFIG_LCD_RPORTRAIT)
+		putarea(0, yres - 1, 0, xres - 1, RED);
+	#else
+		putarea(0, xres - 1, 0, yres - 1, RED);
+	#endif
+	/* fill square with side  = OFFSET */
+	putarea(0, OFFSET, 0, OFFSET, WHITE);
+}
+
+/* Draws horizontal line at a point */
+static void test_put_run(void)
+{
+	int fd = 0;
+	int p = 0;
+	int i;
+	char port[20] = { '\0' };
+	sprintf(port, LCD_DEV_PATH, p);
+	fd = open(port, O_RDWR | O_SYNC, 0666);
+	if (fd < 0) {
+		printf("ERROR: Failed to open lcd port : %s error:%d\n", port, fd);
+		return -1;
+	}
+	struct lcddev_run_s run;
+	run.planeno = 0;
+	run.col = COLINDEX;
+	run.row = ROWINDEX;
+	run.npixels = NOPIXELS;
+
+	uint8_t spi_data[2 * NOPIXELS + 1];
+	run.data = &spi_data;
+	for (i = 0; i <= (NOPIXELS * 2); i += 2) {
+		spi_data[i + 1] = WHITE & 0X00FF;
+		spi_data[i] = (WHITE & 0xFF00) >> 8;
+	}
+	ioctl(fd, LCDDEVIO_PUTRUN, (unsigned long)(uintptr_t)&run);
+	close(fd);
+}
+
+static void test_clear(void)
+{
+	int fd = 0, p = 0;
+	char port[20] = { '\0' };
+	sprintf(port, LCD_DEV_PATH, p);
+	fd = open(port, O_RDWR | O_SYNC, 0666);
+	if (fd < 0) {
+		printf("ERROR: Failed to open lcd port : %s error:%d\n", port, fd);
+		return -1;
+	}
+	struct fb_videoinfo_s vinfo;
+	ioctl(fd, LCDDEVIO_GETVIDEOINFO, (unsigned long)(uintptr_t)&vinfo);
+	xres = vinfo.xres;
+	yres = vinfo.yres;
+	close(fd);
+	#if defined(CONFIG_LCD_PORTRAIT) || defined(CONFIG_LCD_RPORTRAIT)
+		putarea(0, yres - 1, 0, xres - 1, BLACK);
+	#else
+		putarea(0, xres - 1, 0, yres - 1, BLACK);
+	#endif
+}
+
+static void test_init(void)
+{
+	int ret;
+	int fd = 0;
+	int p = 0;
+	char port[20] = { '\0' };
+	sprintf(port, LCD_DEV_PATH, p);
+	fd = open(port, O_RDWR | O_SYNC, 0666);
+	if (fd < 0) {
+		printf("ERROR: Failed to open lcd port : %s error:%d\n", port, fd);
+		return -1;
+	}
+	ioctl(fd, LCDDEVIO_INIT, &ret);
+	close(fd);
+}
+
+static void test_orientation(void)
+{
+	int fd = 0;
+	int p = 0;
+	char port[20] = { '\0' };
+	sprintf(port, LCD_DEV_PATH, p);
+	fd = open(port, O_RDWR | O_SYNC, 0666);
+	if (fd < 0) {
+		printf("ERROR: Failed to open lcd port : %s error:%d\n", port, fd);
+		return -1;
+	}
+	ioctl(fd, LCDDEVIO_SETORIENTATION, ST7789_RLANDSCAPE);
+
+	test_put_run();
+
+	sleep(1);
+	/* resolution should be swapped now as orientation is changed*/
+	putarea(0, xres - 1, 0, yres - 1, RED);
+	/* fill square with side  = OFFSET */
+	putarea(0, OFFSET, 0, OFFSET, WHITE);
+
+	sleep(1);
+	#if defined(CONFIG_LCD_PORTRAIT)
+		ioctl(fd, LCDDEVIO_SETORIENTATION, ST7789_PORTRAIT);
+	#elif defined(CONFIG_LCD_LANDSCAPE)
+		ioctl(fd, LCDDEVIO_SETORIENTATION, ST7789_LANDSCAPE);
+	#elif defined(CONFIG_LCD_RLANDSCAPE)
+		ioctl(fd, LCDDEVIO_SETORIENTATION, ST7789_RLANDSCAPE);
+	#elif defined(CONFIG_LCD_RPORTRAIT)
+		ioctl(fd, LCDDEVIO_SETORIENTATION, ST7789_RPORTRAIT);
+	#else
+		ioctl(fd, LCDDEVIO_SETORIENTATION, ST7789_LANDSCAPE);
+	#endif
```

Comment:
Because of this, this lcd example becomes ST7789 specific example. Is there anyway to use lcd general values?

Reply:
We can have enum of orientaion in lcd_dev.h like
enum
{
   LCD_PORTRAIT,
   LCD_LANDSCAPE,
} lcd_orientation;
and send LCD_PORTRAIT to ioctl, and driver will call ST7789_PORTRAIT if LCD_PORTRAIT is sent.
Since there is need of dynamic orientation required through ioctl, we can't just use config for it.
Config is used for setting initial orientation at time of initialization or revert back to original orientation any time.
If there is a better way, please suggest.

Reply:
> We can have enum of orientaion in lcd_dev.h like
enum
{
LCD_PORTRAIT,
LCD_LANDSCAPE,
} lcd_orientation;
and send LCD_PORTRAIT to ioctl, and driver will call ST7789_PORTRAIT if LCD_PORTRAIT is sent.

This looks enough.